opentelemetry-sdk/src/testing/trace/in_memory_exporter.rs

```diff
@@ -0,0 +1,146 @@
+use crate::export::trace::{ExportResult, SpanData, SpanExporter};
+use futures_util::future::BoxFuture;
+use opentelemetry::trace::{TraceError, TraceResult};
+use std::sync::{Arc, Mutex};
+
+/// An in-memory span exporter that stores span data in memory.
+///
+/// This exporter is useful for testing and debugging purposes. It stores
+/// metric data in a `Vec<SpanData>`. Metrics can be retrieved
+/// using the `get_finished_spans` method.
+/// # Example
+/// ```
+///# use opentelemetry::trace::{SpanKind, TraceContextExt};
+///# use opentelemetry::{global, trace::Tracer, Context};
+///# use opentelemetry_sdk::propagation::TraceContextPropagator;
+///# use opentelemetry_sdk::runtime;
+///# use opentelemetry_sdk::testing::trace::InMemorySpanExporterBuilder;
+///# use opentelemetry_sdk::trace::{BatchSpanProcessor, TracerProvider};
+///
+///# #[tokio::main]
+///# async fn main() {
```

Comment:
Is the line below needed?  It doesn't appear this test uses context propagation...